src/main/java/com/exadel/discountwebapp/statistics/service/StatisticsService.java

```diff
@@ -0,0 +1,139 @@
+package com.exadel.discountwebapp.statistics.service;
+
+import com.exadel.discountwebapp.discount.repository.DiscountRepository;
+import com.exadel.discountwebapp.statistics.dto.*;
+import com.exadel.discountwebapp.userdiscount.repository.UserDiscountRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+@RequiredArgsConstructor
+@Service
+public class StatisticsService {
+
+    private final UserDiscountRepository userDiscountRepository;
+    private final DiscountRepository discountRepository;
+
+    private static final String OTHERS = "Others";
+
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public SummaryStatisticsDTO getStats(LocalDateTime dataFrom, LocalDateTime dataTo) {
+        Map<String, Long> usersDiscountsStats = getDataActivatedDiscountPerUsers(dataFrom, dataTo);
+        Map<String, Long> categoryDiscountsStats = getDataActivatedDiscountPerCategory(dataFrom, dataTo);
+        Map<String, Long> categoryVendorStats = getDataActivatedDiscountPerVendor(dataFrom, dataTo);
+        Map<String, Long> discountViewingStats = getDiscountPerViewing(dataFrom, dataTo);
+        return new SummaryStatisticsDTO(usersDiscountsStats, categoryDiscountsStats, categoryVendorStats, discountViewingStats);
+    }
+
+    public Map<String, Long> getDataActivatedDiscountPerUsers(LocalDateTime dataFrom, LocalDateTime dataTo) {
+        var userData = userDiscountRepository.getUserDiscountStatistics(dataFrom, dataTo);
+        Map<String, Long> result = new HashMap<>();
+        for (UserDTO elem : userData) {
+            result.put(String.format("%s %s, %s", elem.getFirstName(), elem.getLastName(), elem.getEmail()), elem.getQuantity());
```

Comment:
Let's use some class as a key for the map instead of String

Reply:
fixed